src/pineko/check.py

```diff
@@ -92,31 +93,100 @@ def is_fonll_b(fns, lumi):
     return False
 
 
-def contains_fact(grid):
-    """Check whether factorization scale-variations are available in the pineappl grid.
+def contains_fact(grid, max_as, max_al):
+    """Check whether factorization scale-variations are available in the pineappl grid for the requested order.
 
     Parameters
     ----------
         grid: pineappl.grid.Grid
             Pineappl grid
+        max_as: int
+            max as order
+        max_al: int
+            max al order
+    Returns
+    -------
+        : bool
+        is fact scale-variation available for as
+        : bool
+        is fact scale-variation available for al
     """
-    order_list = [order.as_tuple() for order in grid.orders()]
+    order_array = np.array([order.as_tuple() for order in grid.orders()])
+    order_mask = pineappl.grid.Order.create_mask(grid.orders(), max_as, max_al)
+    order_list = order_array[order_mask]
+    as_orders = []
+    al_orders = []
     for order in order_list:
-        if order[-1] != 0:
-            return
-    raise ValueError("Factorization scale variations are not available for this grid")
-
-
-def contains_ren(grid):
+        as_orders.append(order[0])
+        al_orders.append(order[1])
+    min_as = min(as_orders)
+    min_al = min(al_orders)
+    order_as_is_present = False
+    order_al_is_present = False
+    sv_as_present = False
+    sv_al_present = False
+    for order in order_list:
+        # fact sv starts at NLO with respect to the first non zero order
+        if order[0] == min_as + 1:
+            order_as_is_present = True
+            if order[-1] != 0:
```

Comment:
since you said you like the idea of doing a stricter test and this I really meant here: we can exactly predict the necessary powers of logs: at each order you need to have exactly `0..d`  logs, where d is the difference in powers of coupling; e.g. at NLO you need 0,1 logs

Reply:
In practice at the moment the test relies on the fact that a grid can either have the scale variations or not and in the former case the grid would have the scale-variations for all the (available) orders. But of course this is probably not guaranteed

Reply:
I agree the mixed case would be complicated (but that is, if at all, a long term problem) ... again it boils down to the question on [how strict we want to be](https://github.com/N3PDF/pineko/pull/33#issuecomment-1185519566) ... for this let's definitely ask for an opinion to @AleCandido - just to say, your current version is in between the two possibilities I said (and of course viable)

Reply:
I guess in the long term, the "very strict" might be useful. We can even tune according to feedbacks, and expose flags/options to make customization available to the user. From very detailed option to something like `--strictness 3`.

Reply:
In the short term the answer is: close this as soon as possible, it's not a priority, while we're a bit in a hurry for theory covmat. If we have spare time at the moment (meaning "all fktables/fits running, I can't do anything to speed up"), maybe you (@andreab1997) can take over @felixhekhorn task of checking different SV methods.
My feeling is that is a black hole of time, and we'll get almost nothing good, so we should limit the resources dedicated to it. On the other hand, the only good way to close this long-lasting task is to come with a theory covmat paper @NNLO. And if we want a paper, it wouldn't be a bad thing to have a *sufficient* understanding of what's going on - the *sufficient* concept will change with time, of course (and it is already changed).

Reply:
Ok I agree. Then my proposal is to leave the test as it is now and, for the moment, to rely on the fact that the grids can only either have the scale variations or not. At least in this way I can compute the fktables now. Then I will update the check and for sure I would do what you said (@AleCandido0, i.e. asking the user for a `strictness` parameter. Do you agree?

Reply:
I agree. About the **then** part we will rediscuss *then* :)